File: Data Structures & Algorithms/maximum-subarray/submission-1.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int m = nums.size();
        vector<int> dp (m+1 , 0);
        int maxSum = nums[0];
        for(int i = 1;i<=m;i++)
        {
            dp[i]=max(nums[i-1],dp[i-1]+nums[i-1]);
            if (i == 1) maxSum = dp[i];
            else maxSum = max(maxSum, dp[i]);
        }
        return maxSum;


    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix=[]
        t=1
        for i in nums:
            prefix.append(t)
            t*=i
        res=[]
        pt=1
        postfix=[]
        for i in range(len(nums)-1,-1,-1):
            postfix.append(pt)
            pt*=nums[i]
        postfix.reverse()
        for i in range(len(nums)):
            res.append(prefix[i]*postfix[i])
        return res

        
            
            
            